libs/sdk-ui-tests-e2e/scenarios/src/components/Scenarios/Visualizations/PivotTable/Sizing/PivotTableComplexResetScenario.tsx

```diff
@@ -0,0 +1,218 @@
+    ILocatorItem,
+    IMeasureLocatorItem,
+    IAttribute,
+    attributeLocalId,
+    IMeasure,
+    measureLocalId,
+    newAttributeLocator,
+} from "@gooddata/sdk-model";
+
+const measures = [ReferenceMd.Amount];
+const attributes = [ReferenceMd.Product.Name];
+const columns = [ReferenceMd.Region];
+
+export const attributeColumnWidthItem = (attribute: IAttribute, width: number) => {
+    return {
+        attributeColumnWidthItem: {
+            width: { value: width },
+            attributeIdentifier: attributeLocalId(attribute),
+        },
+    };
+};
+
+export const measureColumnWidthItemSimple = (
+    measure: IMeasure,
+    width: number,
+    obj?: IAttributeLocatorItem[],
+): IMeasureColumnWidthItem => {
+    const measureColumnWidthItemHolder: IMeasureColumnWidthItem["measureColumnWidthItem"] = {
+        width: { value: width },
+        locators: [
+            {
+                measureLocatorItem: {
+                    measureIdentifier: measureLocalId(measure),
+                },
+            },
+        ],
+    };
+
+    if (obj) {
+        measureColumnWidthItemHolder.locators.splice(
+            -2,
+            0,
+            ...obj.map((attributeLocatorItem) => ({
+                ...attributeLocatorItem,
+            })),
+        );
+    }
+
+    return { measureColumnWidthItem: measureColumnWidthItemHolder };
+};
+
+export const allMeasureColumnWidthItem = (width: number) => {
+    return {
+        measureColumnWidthItem: {
+            width: { value: width },
+        },
+    };
+};
+
+const attributeWidth = (width: number) => attributeColumnWidthItem(attributes[0], width);
+
+const allMeasureWidth = (width: number) => allMeasureColumnWidthItem(width);
+
+const areLocatorsEqual = (locator1: ILocatorItem[], locator2: ILocatorItem[]) => {
+    return (
+        (locator1[0] as IAttributeLocatorItem).attributeLocatorItem.element ===
+            (locator2[0] as IAttributeLocatorItem).attributeLocatorItem.element &&
+        (locator1[1] as IMeasureLocatorItem).measureLocatorItem.measureIdentifier ===
+            (locator2[1] as IMeasureLocatorItem).measureLocatorItem.measureIdentifier
+    );
+};
+
+const isSameWidthItem = (item: any, newItem: any) => {
+    if (isAttributeColumnWidthItem(item) && isAttributeColumnWidthItem(newItem)) {
+        return (
+            item.attributeColumnWidthItem.attributeIdentifier ===
+            newItem.attributeColumnWidthItem.attributeIdentifier
+        );
+    }
+
+    if (isMeasureColumnWidthItem(item) && isMeasureColumnWidthItem(newItem)) {
+        return areLocatorsEqual(
+            item.measureColumnWidthItem.locators as ILocatorItem[],
+            newItem.measureColumnWidthItem.locators as ILocatorItem[],
+        );
+    }
+
+    if (isAllMeasureColumnWidthItem(item) && isAllMeasureColumnWidthItem(newItem)) {
+        return true;
+    }
+
+    return false;
+};
+
+const EastCoastUri = "/gdc/md/frho3i7qc6epdek7mcgergm9vtm6o5ty/obj/1083/elements?id=460488";
```

Comment:
Can we somehow build this from the workspace? So that if we change workspace for some reason this does not break?

Reply:
fixed